### PR TITLE
revolt-desktop: darwin support

### PR DESCRIPTION
**Pull request by magistau**
## Description of changes

Added darwin support. Source-based build is not an option yet (see #317842)

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by drupol — DISMISSED** [review]
Would it be possible to:

1. Use `stdenvNoCC` ?
2. Use only one occurrence of `stdenvNoCC` instead of 2 ? (there are many things in common that can be put in common)

**Review by heyimnova — APPROVED** [review]

**Review by JohnRTitor — COMMENTED** [review]
Do we need to use overrideAttrs here? Can we not do it like [warp-terminal](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/by-name/wa/warp-terminal/package.nix)? (maybe not the best example)

If you have time please move this to `by-name` as well!

**Comment by thecaralice** [comment]
> Do we need to use overrideAttrs here? Can we not do it like [warp-terminal](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/by-name/wa/warp-terminal/package.nix)? (maybe not the best example)

This is in contrary with @drupol's request; to be honest, I'm in favor of not duplicating things

> If you have time please move this to `by-name` as well!

Will do

File: pkgs/by-name/re/revolt-desktop/default.nix
Comment (JohnRTitor) [inline pkgs/by-name/re/revolt-desktop/default.nix:1]: Renamed file should be `package.nix`

**Review by JohnRTitor — DISMISSED** [review]

File: pkgs/by-name/re/revolt-desktop/default.nix
Comment (JohnRTitor) [reply pkgs/by-name/re/revolt-desktop/default.nix:1]: Did you drop the default.nix accidentally instead of renaming it properly?

Comment (thecaralice) [reply pkgs/by-name/re/revolt-desktop/default.nix:1]: Did a `git commit -a` without thinking about `package.nix` being untracked. Should've got some sleep I guess.

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 320128` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>revolt-desktop</li>
  </ul>
</details>

**Pull request merged**